### ISSUE TITLE
Change rotation behavior?

### ISSUE THREAD

**Issue by nazar-pc** [solvespace/solvespace#325]
### System information

SolveSpace version: 2.3

Operating system: Ubuntu 18.10

### Additional information

I'm new to solvespace, but would say that the way rotation is implemented in solvespace is at very least strange. Every time I try to rotate model with mouse, it does almost exactly the opposite of behavior I'm used to in other programs like free online Tinkercad. To be clear, I'm not talking about mouse/keyboard keys, but rather about rotation process. I almost want to invert it, at least partially.

Is there any way to configure how rotation happens? I didn't find any relevant issues or settings in documentation or config file.

**Comment by whitequark** [comment]
There is currently no way to configure the controls. This has been discussed before; pretty much every CAD program uses its own controls, so modifying ours would inevitably displease everyone except people who use one particular CAD program. It seems pointless.

**Comment by nazar-pc** [comment]
I've tried FreeCAD briefly, it had a few hardcoded presets for this behavior that one can chose from. Do you think something like this can be added to SolveSpace?

**Comment by whitequark** [comment]
Not easily, the code that handles controls is not currently written in a way that is amenable to this sort of customization. It'd require a major refactor.

**Comment by nazar-pc** [comment]
I believe it would be useful for many people. So if there is a chance someone will tackle this in future, would be nice to label this as feature request.

**Comment by notnot** [comment]
The rotation controls are also counter-intuitive to me. It is almost trial and error to get the view i am after...

**Comment by whitequark** [comment]
Does it help if you tick the "use turntable mouse navigation" checkbox on master?

**Comment by nazar-pc** [comment]
Unfortunately no, it is even more confusing with that option, can't imagine someone would actually want that.
The primary issue I have is that any circular-like motion turns the model in the opposite direction to what I expect. I've got used to most of other things.

**Comment by hinell** [comment]
Thumbing up. Even with turntable is enabled it still feels quirky compared to blender.
I think rotation around the `X` axis is not correct. 

https://user-images.githubusercontent.com/8136158/151233874-cc101e33-cbdb-45aa-a91b-47adf80d08e6.mp4


https://user-images.githubusercontent.com/8136158/151233878-34df3b59-ebba-4831-9661-42ed34bdf304.mp4

**Comment by Symbian9** [comment]
> solvespace-rotation.mp4

@hinell, before rotate select XYZ-normal origin point from active `g001-references` (or any other point to be used as rotation origin) and click <kbd>F4</kbd> (or go to menu *"View > Center View At Point"*) — then retry rotation behavior.

**Comment by phkahler** [comment]
@hinell   Imagine the model is sitting inside a glass ball. Moving the mouse UP is like rolling that ball upward by pushing your hand (or mouse) across it. Another analogy is to think of a trackball, and the model is inside the trackball. However you want to think of it, the rotation is rotating the model, not the camera. Well, it actually is moving the camera because none of the coordinates in the model change, but it's supposed to feel like moving the object. Does any of this explanation help?

**Comment by Symbian9** [comment]
> However you want to think of it, the rotation is rotating the model, not the camera.

@phkahler, no, you missed the point where @hinell told that he *"think rotation around the `X` axis is not correct."* — problem here is that selecting/setting origin of 3D view rotation is not well documented in SolveSpace docs.

In [LeoCAD](https://github.com/leozide/leocad) app there is excellent UX on 3D view rotating with many various methods.
